Add 'disabler' argument to functions returning not-found objects

xclaesse opened the pull request:
When dependency(), find_library() or find_program() return a not-found
object and use_disabler is true, they return a Disabler object instead.

xclaesse commented:
@jpakkane any objection on this feature-wise ?

xclaesse commented:
Yeah, naming stuff is often the hardest part. I have no good answer here :/

nirbheek commented:
When there's no way to have abbreviated API that also makes sense, it's best to put something that forces people to look up the documentation to understand what it does, but is also easy to remember once you've looked it up.

To that end, my vote goes to `disabler: true`.

jpakkane commented:
Sounds good to me.

xclaesse commented:
PR updated with the new name, added to python.find_installation(), added release notes, updated version to 0.49, and fixed new feature usage (@disablerIfNotFound must be after @FeatureNewKwargs and before @permittedKwargs).

jpakkane commented:
CI runs are not starting. :( Is Github still broken or has the outage wiped our configurations?

xclaesse commented:
CI failure is unrelated, should be good to merge.
